datasets/monero/monero.py

```diff
@@ -0,0 +1,198 @@
+
+        urls_to_download = {
+            "monero": _URL + _MONERO_FILE,
+            "cardiology": _URL + _CARDIOLOGY_FILE,
+            "diabetes": _URL + _DIABETES_FILE,
+            "endocrinology": _URL + _ENDOCRINOLOGY_FILE,
+        }
+
+        downloaded_files = dl_manager.download(urls_to_download)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"filepath": downloaded_files["endocrinology"]},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"filepath": downloaded_files["cardiology"]},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"filepath": downloaded_files["diabetes"]},
+            ),
+        ]
+
+    def _generate_examples(self, filepath):
+        """ Yields examples. """
+
+        logging.info("⏳ Generating examples from = %s", filepath)
+        with open(filepath, encoding="utf-8") as f:
+            guid = 0
+            words = []
+            ner = []
+            start = []
+            end = []
+            sent = ""
+            has_started = False
+            for line in f:
+                if line == "\n":
+                    if sent:
+                        words.append(sent)
+                        yield guid, {
+                            "id": str(guid),
+                            "words": words,
+                            "start": start,
+                            "end": end,
+                            "ner": ner,
+                        }
+                        guid += 1
+                        words = []
+                        start = []
+                        end = []
+                        ner = []
+                        sent = ""
+                else:
+                    # ronec words are tab separated
+                    line = line.replace("\n", "")
+                    splits = line.split("\t")
+                    sent += splits[1] + " "
+
+                    if splits[2].startswith("O") and not has_started:
+                        continue
+                    elif splits[2].startswith("B-"):
+                        begin = len(sent) - len(splits[1])
+                        last = len(sent)
+                        label = splits[2][2:]
+                        has_started = True
+                    elif splits[2].startswith("I-"):
+                        last = len(sent)
+                    elif splits[2].startswith("O") and begin:
+                        ner.append(label)
+                        start.append(begin)
+                        end.append(last)
+                        has_started = False
```

Comment:
You shouldn't remove the IOB tags.

You can get some inspiration from conll2003: https://github.com/huggingface/datasets/blob/master/datasets/conll2003/conll2003.py